File: backend/app/apis/system/id_center.py
# -*- coding: utf-8 -*-
# @author: xiaobai
"""
个人中心模块 (Identity Center)

提供个人中心相关功能，包括：
- 获取个人信息
- 更新个人信息
- 更新头像
- 更新手机号
- 更新邮箱
- 修改密码
- 获取个人菜单权限
"""
from app.corelibs.custom_router import APIRouter
from app.schemas.system.user import (
    UserResetPwd,
    UserProfileUpdate,
    UserAvatarUpdate,
    UserEmailUpdate,
    UserPhoneUpdate
)
from app.services.system.user import UserService
from app.utils.response import HttpResponse

router = APIRouter()


@router.post('/getUserInfo', description="获取个人信息")
async def get_user_info():
    """
    获取当前登录用户的个人信息
    
    :return: 用户详细信息
    """
    user_info = await UserService.get_user_info_by_token()
    return await HttpResponse.success(user_info)


@router.post('/updateProfile', description="更新个人信息")
async def update_profile(params: UserProfileUpdate):
    """
    更新个人信息
    
    支持更新：昵称、邮箱、头像、个人描述
    
    :param params: 个人信息参数
    :return: 更新后的用户信息
    """
    result = await UserService.update_profile(params)
    return await HttpResponse.success(result)


@router.post('/updateAvatar', description="更新头像")
async def update_avatar(params: UserAvatarUpdate):
    """
    更新用户头像
    
    :param params: 头像参数
    :return: 更新后的用户信息
    """
    result = await UserService.update_avatar(params)
    return await HttpResponse.success(result)


@router.post('/updateEmail', description="更新邮箱")
async def update_email(params: UserEmailUpdate):
    """
    更新用户邮箱
    
    :param params: 邮箱参数
    :return: 更新后的用户信息
    """
    result = await UserService.update_email(params)
    return await HttpResponse.success(result)


@router.post('/updatePhone', description="更新手机号")
async def update_phone(params: UserPhoneUpdate):
    """
    更新用户手机号
    
    :param params: 手机号参数
    :return: 更新后的用户信息
    """
    result = await UserService.update_phone(params)
    return await HttpResponse.success(result)


@router.post('/resetPassword', description="修改密码")
async def reset_password(params: UserResetPwd):
    """
    修改个人密码

    :param params: 密码参数
    :return: 修改结果
    """
    await UserService.reset_password(params)
    return await HttpResponse.success()


@router.post('/getLoginRecords', description="获取登录记录")
async def get_login_records(page: int = 1, page_size: int = 10):
    """
    获取当前用户的登录记录

    :param page: 页码，默认为1
    :type page: int
    :param page_size: 每页数量，默认为10
    :type page_size: int
    :return: 登录记录列表
    """
    result = await UserService.get_login_records(page, page_size)
    return await HttpResponse.success(result)


@router.post('/getMenu', description="获取个人菜单权限")
async def get_menu():
    """
    获取当前用户的菜单权限
    
    :return: 菜单列表
    """
    user_info = await UserService.get_menu_by_token()
    return await HttpResponse.success(user_info)

File: frontend/src/api/useSystemApi/idCenter.ts
import request from '/@/utils/request';

/**
 * 个人中心接口
 * @method getUserInfo 获取个人信息
 * @method updateProfile 更新个人信息
 * @method updateAvatar 更新头像
 * @method updatePhone 更新手机号
 * @method updateEmail 更新邮箱
 * @method resetPassword 修改密码
 * @method getMenu 获取个人菜单权限
 */
export function useIdCenterApi() {
  return {
    // 获取个人信息
    getUserInfo: () => {
      return request({
        url: '/profile/getUserInfo',
        method: 'POST',
        data: {},
      });
    },
    // 更新个人信息（批量）
    updateProfile: (data?: object) => {
      return request({
        url: '/profile/updateProfile',
        method: 'POST',
        data,
      });
    },
    // 更新头像
    updateAvatar: (data?: object) => {
      return request({
        url: '/profile/updateAvatar',
        method: 'POST',
        data,
      });
    },
    // 更新手机号
    updatePhone: (data?: object) => {
      return request({
        url: '/profile/updatePhone',
        method: 'POST',
        data,
      });
    },
    // 更新邮箱
    updateEmail: (data?: object) => {
      return request({
        url: '/profile/updateEmail',
        method: 'POST',
        data,
      });
    },
    // 修改密码
    resetPassword: (data?: object) => {
      return request({
        url: '/profile/resetPassword',
        method: 'POST',
        data,
      });
    },
    // 获取个人菜单权限
    getMenu: () => {
      return request({
        url: '/profile/getMenu',
        method: 'POST',
        data: {},
      });
    },
    // 获取登录记录
    getLoginRecords: (data?: object) => {
      return request({
        url: '/profile/getLoginRecords',
        method: 'POST',
        data,
      });
    },
  };
}
